### PR TITLE
TST: test for intentionally emitted warnings, avoid some unintentional ones

### PR DESCRIPTION
**Pull request by jbrockmendel**
Selected by looking through Travis logs and picking out warnings produced internally.

### CODE REVIEW

**Comment by bashtage** [comment]
LGTM

File: statsmodels/tools/tests/test_grouputils.py [hunk 1]
```diff
@@ -314,24 +314,24 @@ def test_dummy_sparse():
...
     assert isinstance(indi, sparse.csr.csr_matrix)
     result = indi.todense()
-    expected = np.matrix([[1, 0, 0],
-                          [1, 0, 0],
-                          [0, 0, 1],
-                          [0, 1, 0],
-                          [0, 1, 0],
-                          [0, 0, 1],
-                          [1, 0, 0]], dtype=np.int8)
+    expected = np.array([[1, 0, 0],
```
Comment (josef-pkt) [inline statsmodels/tools/tests/test_grouputils.py]: in my scipy version `todense` returns a matrix and not an array. 
np.matrix is intentional, or at least it was at the time

Comment (jbrockmendel) [reply statsmodels/tools/tests/test_grouputils.py]: what scipy version?

**Comment by bashtage** [comment]
Py 27 has always been strange about repeating warnings.  Perhaps just use warnings.record here and add a TODO note to use the correct check once Python 2.7 is removed.

File: statsmodels/tools/tests/test_grouputils.py (hunk 1 above)
Comment (josef-pkt) [reply statsmodels/tools/tests/test_grouputils.py]: I'm using 0.18, but the current docs also say matrix is returned
https://docs.scipy.org/doc/scipy/reference/generated/scipy.sparse.csr_matrix.todense.html

**Comment by bashtage** [comment]
SciPy sparse still uses matrix.  It is a bit of an annoyance since matrix is pending deprecation or future warning now.  I think the path forward hasn’t been worked out, so it might stay this way for a while. 

IMO matrix should be removed where possible, and sparse arrays can be directly exported to arrays using toarray.

**Comment by josef-pkt** [comment]
But then we leave the matrix use alone for now and do a special PR before scipy.sparse switches.
i.e. don't change the unit test, but review usage of matrices after todense in the code.

In a file search I only saw todense use in a few places outside unit tests

**Comment by jbrockmendel** [comment]
I'm pretty happy to stick a pin in this and revisit after py27 is dropped.

**Comment by bashtage** [comment]
Any interest in getting this merged?  Just a could be line reversions away.

**Comment by jbrockmendel** [comment]
I'd rather leave it until after py2 is dropped.

**Comment by jbrockmendel** [comment]
@bashtage py2-sensitive changes reverted as suggested

**Pull request merged**